src/clean.c

```diff
@@ -189,13 +189,39 @@ static bool is_all_digits(const char *s)
 	return true;
 }
 
+static bool is_all_xdigits(const char *s)
+{
+	for (; *s; s++) {
+		if (!isxdigit(*s)) {
+			return false;
+		}
+	}
+	return true;
+}
+
+static bool is_hashed_manifest(const char UNUSED_PARAM *dir, const struct dirent *entry)
+{
+	static const char prefix[] = "Manifest.";
+	static const size_t prefix_len = sizeof(prefix) - 1;
+	if (strncmp(entry->d_name, prefix, prefix_len)) {
+		return false;
+	}
+	/* grab hash suffix by splitting on last '.' then iterating past it (+ 1) */
+	return is_all_xdigits(strrchr(entry->d_name, '.') + 1);
+}
+
 static bool is_manifest(const char UNUSED_PARAM *dir, const struct dirent *entry)
 {
 	static const char prefix[] = "Manifest.";
 	static const size_t prefix_len = sizeof(prefix) - 1;
 	return !strncmp(entry->d_name, prefix, prefix_len);
 }
 
+static bool is_manifest_type(const char UNUSED_PARAM *dir, const struct dirent *entry)
+{
+	return is_manifest(dir, entry) || is_hashed_manifest(dir, entry);
```

Comment:
Isn't all hashed_manifests also manifests? I think clean --all is already cleaning hashed manifests

Reply:
Ah, yep, fixed and pushed.